### PR TITLE
ref(discover): Move to using json for range functions

### PR DESCRIPTION
**Pull request by Zylphrex**

### CODE REVIEW

File: tests/sentry/api/test_event_search.py [hunk 1]
```diff
@@ -2391,8 +2391,22 @@ def test_percentile_range(self):
         result = resolve_field_list(fields, eventstore.Filter())
         assert result["aggregations"] == [
             [
-                u"quantileIf(0.50)(duration,and(greaterOrEquals(timestamp,toDateTime('2020-05-01T01:12:34')),less(timestamp,toDateTime('2020-05-03T06:48:57'))))",
-                None,
+                # u"quantileIf(0.50)(duration,and(greaterOrEquals(timestamp,toDateTime('2020-05-01T01:12:34')),less(timestamp,toDateTime('2020-05-03T06:48:57'))))",
```
Comment (dashed) [inline tests/sentry/api/test_event_search.py:2394]: stray comment

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -1805,30 +1805,48 @@ def validate_argument_count(self, field, arguments):
...
-                u"quantileIf({percentile:.2f})({column},and(greaterOrEquals(timestamp,toDateTime('{start}')),less(timestamp,toDateTime('{end}'))))",
-                None,
+                u"quantileIf({percentile:.2f})",
+                [
+                    ArgValue("column"),
+                    [
+                        "and",
+                        [
+                            ["lessOrEquals", [["toDateTime", [ArgValue("start")]], "timestamp"]],
+                            ["greater", [["toDateTime", [ArgValue("end")]], "timestamp"]],
```
Comment (markstory) [inline src/sentry/api/event_search.py:1822]: Did these invert to appease the snuba query syntax? Flipping the function names doesn't make sense to me as the range is now 'less than or equals to start, and greater than the end' which seems like it is not counting the selected window.

Comment (Zylphrex) [reply src/sentry/api/event_search.py:1822]: Unfortunately, yes. If the condition was written as `["greaterOrEquals", ["timestamp", ["toDateTime", [ArgValue("start")]]]]`, snuba special cases this as `greaterOrEquals(timestamp(toDatetime(start)))` which is not what we want, and inverting these conditions allows us to specify them in a way snuba expects.

Comment (wmak) [reply src/sentry/api/event_search.py:1822]: Could we add a comment explaining this?

**Review by wmak — APPROVED** [review]

File: src/sentry/api/event_search.py (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/api/event_search.py:1840]: Added comments in 2f813b4

**Review by markstory — APPROVED** [review]

**Pull request merged**